### PR TITLE
Case history updates

### PR DESCRIPTION
**Pull request by NickGrantONS**
# Motivation and Context

The case history should show an event when a case is included in a print file. We need to send events back to the Case Processor service from the Action Scheduler when an action rule is triggered, so that we can record an event against the cases which were selected.

# What has changed

Created DTO for message sent back to Case. Created new queue and exchange. Created builder for message. Build and send message when we send message onto Print Service.

# How to test?

Load a sample. Create an action plan & rule. You should see messages on the `action.events` queue appear when the rule executes.

# Links

Trello: https://trello.com/c/seWLHMKQ

# Screenshots (if appropriate):

### CODE REVIEW

**Comment by NickGrantONS** [comment]
[RMC-212 Action Scheduler - Update case history when added to print file (8)**](https://trello.com/c/seWLHMKQ/881-rmc-212-action-scheduler-update-case-history-when-added-to-print-file-8)

File: src/test/java/uk/gov/ons/census/action/schedule/ActionRuleProcessorTest.java [hunk 1]
```diff
@@ -99,92 +117,63 @@ public void testExecuteAllCasesField() {
...
         .thenReturn(new uk.gov.ons.census.action.model.dto.instruction.field.ActionInstruction());
 
-    // when
     ActionRuleProcessor actionRuleProcessor =
         new ActionRuleProcessor(
-            actionRuleRepo, caseRepository, actionInstructionBuilder, null, rabbitFieldTemplate);
+            actionRuleRepo,
+            actionInstructionBuilder,
+            printFileDtoBuilder,
+            null,
```
Comment (davidmort) [inline src/test/java/uk/gov/ons/census/action/schedule/ActionRuleProcessorTest.java]: looks like ur missing the printCaseSelectedBuilder argument here

Comment (NickGrantONS) [reply src/test/java/uk/gov/ons/census/action/schedule/ActionRuleProcessorTest.java]: fixed

File: src/main/java/uk/gov/ons/census/action/builders/QidUacBuilder.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import uk.gov.ons.census.action.model.entity.UacQidLink;
+import uk.gov.ons.census.action.model.repository.UacQidLinkRepository;
+
+@Component
+public class QidUacBuilder {
+  private static final int NUM_OF_UAC_IAC_PAIRS_NEEDED_BY_A_WALES_INITIAL_CONTACT_QUESTIONNAIRE = 2;
+  private static final int NUM_OF_UAC_IAC_PAIRS_NEEDED_FOR_SINGLE_LANGUAGE = 1;
+  private static final String WALES_IN_ENGLISH_QUESTIONNAIRE_TYPE = "02";
+  private static final String WALES_IN_WELSH_QUESTIONNAIRE_TYPE = "03";
+  public static final String HOUSEHLD_INITIAL_CONTACT_QUESTIONNIARE_TREATMENT_CODE_PREFIX = "HH_Q";
```
Comment (hbrace) [inline src/main/java/uk/gov/ons/census/action/builders/QidUacBuilder.java]: Feels like I'm being very picky but do we care about the misspelled QUESTIONNAIRE?

File: src/main/java/uk/gov/ons/census/action/builders/ActionInstructionBuilder.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    uk.gov.ons.census.action.model.dto.instruction.field.ActionRequest actionRequest =
+        new uk.gov.ons.census.action.model.dto.instruction.field.ActionRequest();
+    actionRequest.setActionId(UUID.randomUUID().toString());
+    actionRequest.setResponseRequired(false);
+    actionRequest.setActionPlan(actionRule.getActionPlan().getId().toString());
+    actionRequest.setActionType(actionRule.getActionType().toString());
+    actionRequest.setAddress(actionAddress);
+    actionRequest.setCaseId(caze.getCaseId().toString());
+    actionRequest.setPriority(uk.gov.ons.census.action.model.dto.instruction.field.Priority.MEDIUM);
+    actionRequest.setCaseRef(Long.toString(caze.getCaseRef()));
```
Comment (hbrace) [inline src/main/java/uk/gov/ons/census/action/builders/ActionInstructionBuilder.java]: Should this be an int?

File: src/main/java/uk/gov/ons/census/action/builders/QidUacBuilder.java (hunk 2 above)
Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/census/action/builders/QidUacBuilder.java]: this has been merged in from @LukeLoze1 's branch but I guess I could fix it here

File: src/main/java/uk/gov/ons/census/action/builders/ActionInstructionBuilder.java (hunk 3 above)
Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/census/action/builders/ActionInstructionBuilder.java]: Action Scheduler is still using a `long` to store its caseref. I guess we could change it as part of this PR

Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/census/action/builders/ActionInstructionBuilder.java]: my bad... it has been changed to an 'int' should be fixed now

**Pull request merged**